lib/Transforms/Scalar/ScalarReplAggregatesHLSL.cpp

```diff
@@ -6344,6 +6347,8 @@ void SROA_Parameter_HLSL::createFlattenedFunction(Function *F) {
       }
 
       flatArg->replaceAllUsesWith(Arg);
+      if (isa<Instruction>(flatArg))
```

Comment:
This does not fix any bug, but there's no need to keep those temp alloca's around once replaced with the new function's argument. They probably got eliminated by dead code removal later on.